iroha-balancer-client/src/main/kotlin/soramitsu/irohautils/balancer/client/service/IrohaBalancerClientService.kt

```diff
@@ -0,0 +1,101 @@
+package soramitsu.irohautils.balancer.client.service
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.rabbitmq.client.BuiltinExchangeType
+import com.rabbitmq.client.Connection
+import com.rabbitmq.client.ConnectionFactory
+import com.rabbitmq.client.impl.DefaultExceptionHandler
+import iroha.protocol.TransactionOuterClass
+import jp.co.soramitsu.iroha.java.Utils
+import mu.KLogging
+import soramitsu.irohautils.balancer.client.config.RMQConfig
+import java.io.Closeable
+import kotlin.system.exitProcess
+
+open class IrohaBalancerClientService @JvmOverloads constructor(
+        val rmqConfig: RMQConfig,
+        private val onRmqFail: () -> Unit = {
+            logger.error { "RMQ failure. Exit." }
+            exitProcess(1)
```

Comment:
When someone will use this library and there will be any trouble with RMQ are you going to shutdown the whole application? It is a bad idea. Let't throw exception upper and let client decide what to do. Maybe we will use some sort of configuration builder which will allow library users set error handler?